### PR TITLE
[Tensor Operants & Prim] Tensor arithmetic operants support right scalar type

### PR DESCRIPTION
**Pull request by jiahy0825**
### PR types
New features

### PR changes
Others

### Describe
本 PR 工作项点：
1. 支持 Tensor +-*/ Scalar 运算符重载
2. 扩充组合算子静态图模式下，full op 支持的数据类型
3. 涉及 Scalar 的四则运算需要额外调用 fill_constant op，因此修改了组合算子单测

Items of this PR:
1. Tensor arithmetic operants support right scalar type, e.g. `Tensor +-*/ Scalar`
2. Support more datatype of full op in prim static mode.
3. Scalar-relevant operants need to invoke fill_constant op, therefore modifying some unit tests of prim.

### CODE REVIEW

**Review by chenwhql — APPROVED** [review]

**Review by zyfncg — APPROVED** [review]

File: python/paddle/fluid/tests/unittests/prim/test_comp_get_grad_op_desc_prim_enabled.py [hunk 1]
```diff
@@ -41,7 +41,14 @@
...
             set(),
             tuple(),
-            ('pow', 'scale', 'elementwise_mul'),
+            (
+                'pow',
+                'fill_constant',
+                'elementwise_mul',
+                'fill_constant',
+                'elementwise_add',
+                'elementwise_mul',
```
Comment (zyfncg) [inline python/paddle/fluid/tests/unittests/prim/test_comp_get_grad_op_desc_prim_enabled.py:50]: 这里是不重复了？

**Review by YuanRisheng — APPROVED** [review]

File: python/paddle/fluid/tests/unittests/prim/test_comp_get_grad_op_desc_prim_enabled.py (hunk 1 above)
Comment (jiahy0825) [reply python/paddle/fluid/tests/unittests/prim/test_comp_get_grad_op_desc_prim_enabled.py:50]: 此处表示的是算子的调用顺序，并未重复

Here represents the order of calling operators.

File: paddle/fluid/prim/api/api.yaml [hunk 2]
```diff
@@ -1,13 +1,14 @@
+- add
+- subtract
+- multiply
+- divide
 - unsqueeze
 - pow
```
Comment (JiabinYang) [inline paddle/fluid/prim/api/api.yaml:6]: maybe we don't need this?

**Review by JiabinYang — COMMENTED** [review]
one comment

File: paddle/fluid/prim/api/api.yaml (hunk 2 above)
Comment (jiahy0825) [reply paddle/fluid/prim/api/api.yaml:6]: 以加法为例，虽然用户不再需要调用 add 函数，可以直接调用 + 运算符。
但是重载后的 + 操作，实际依赖的还是 add 函数，因此需要保留这几项，自动生成 add 函数的声明。

Although users could invoke the `+` operant directly without calling the `add` function, in fact, the overloaded `+` operant relies on the `add` function. Hence, we need to keep these items to generate an `add` declaration automatically.

**Review by JiabinYang — APPROVED** [review]
LGTM

**Pull request merged**